### PR TITLE
Slit Trace Fix

### PR DESCRIPTION
**Pull request by profxj**
Fix to spca definition for yspl.  Now uses ordcen which 
is proper as this is where the Wavelength solution is generated.

### CODE REVIEW

**Comment by profxj** [comment]
@rcooke-ast might check whether a similar fix needs to be in spline or interp.

File: src/artrace.py [hunk 1]
```diff
@@ -1209,7 +1209,13 @@ def model_tilt(slf, det, msarc, censpec=None, maskval=-999999.9,
         elif slf._argflag['trace']['orders']['tilts'].lower() == "spca":
              xspl = np.linspace(0.0, 1.0, msarc.shape[1])
              #yspl = np.append(0.0, np.append(arcdet[np.where(aduse)]/(msarc.shape[0]-1.0), 1.0))
-             yspl = np.append(0.0, np.append(polytilts[arcdet[np.where(aduse)],msarc.shape[1]/2],1.0))
```
Comment (rcooke-ast) [inline src/artrace.py]: I'm pretty sure that this bug is not present in the interp method, and doesn't need to be resolved for the spline method. You might want to check, but I think your yspl method is the same as the more condensed form:
```
ycen = np.diag(polytilts[arcdet[np.where(aduse)], ordcen[arcdet[np.where(aduse)]]])
yspl = np.append(0.0, np.append(ycen, 1.0))
```
but you might want to check... (see line 1149-1150).

**Comment by rcooke-ast** [comment]
There's another "bug" with the spca and spline methods, I believe. Because these methods are a fit to the arc line traces, the end points (in the spectral direction) should be exactly 0.0 and 1.0 at the location where the arc spectrum is extracted. We haven't presently set these boundary conditions, so the wavelength calibration (at least at the ends of the spectrum) will be off by `msarc.shape[0] * (difference from 0.0 or 1.0)` pixels.

The interp method, on the other hand, forces the boundary condition.

File: src/artrace.py (hunk 1 above)
Comment (profxj) [inline src/artrace.py]: I was trying to figure out a Pythonic way to do this! 
Replaced.

**Comment by profxj** [comment]
Should we modify these lines to force the boundary condition?

             zspl[:, 0] = zspl[:, 1] + polytilts[0, :] - polytilts[arcdet[np.where(aduse)[0][0]], :]
             zspl[:, -1] = zspl[:, -2] + polytilts[-1, :] - polytilts[arcdet[np.where(aduse)[0][-1]], :]

**Comment by profxj** [comment]
merging for now

**Comment by rcooke-ast** [comment]
It is in fact these lines that _encourage_ the boundary conditions to be met (see lines 1240-1241):

```
zspl[:, 0] -= zspl[ordcen[0, 0], 0]
zspl[:, -1] = zspl[:, -1] - zspl[ordcen[-1, 0], -1] + 1.0
```

but this does not _enforce_ the boundary conditions.

**Pull request merged**